src/components/filters/MeasureValueFilter/OperatorDropdown.tsx

```diff
@@ -0,0 +1,109 @@
+
+import OperatorDropdownBody from "./OperatorDropdownBody";
+import * as Operators from "./operators";
+import { getOperatorTranslationKey, getOperatorIcon } from "./utils";
+
+interface IOperatorDropdownOwnProps {
+    onSelect: (operator: string) => void;
+    operator: string;
+}
+
+type IOperatorDropdownProps = IOperatorDropdownOwnProps & InjectedIntlProps;
+
+interface IOperatorDropdownState {
+    opened: boolean;
+}
+
+export class OperatorDropdown extends React.PureComponent<IOperatorDropdownProps, IOperatorDropdownState> {
+    public static defaultProps: any = {
+        onSelect: noop,
+        operator: Operators.ALL,
+    };
+
+    constructor(props: any) {
+        super(props);
+
+        this.state = {
+            opened: false,
+        };
+    }
+
+    public render() {
+        const { operator } = this.props;
+        return (
+            <>
+                {this.renderDropdownButton()}
+                {this.state.opened ? (
+                    <OperatorDropdownBody
+                        alignTo={".gd-mvf-operator-dropdown-button"}
+                        onSelect={this.handleOperatorSelected}
+                        selectedOperator={operator}
+                        onClose={this.handleOperatorDropdownClosed}
+                    />
+                ) : null}
+            </>
+        );
+    }
+
+    private renderDropdownButton() {
+        const { intl, operator } = this.props;
+        const { opened } = this.state;
+
+        const title = capitalize(intl.formatMessage({ id: getOperatorTranslationKey(operator) }));
+
+        const buttonClasses = classNames(
+            "gd-mvf-operator-dropdown-button",
+            "s-mvf-operator-dropdown-button",
+            `s-mvf-operator-dropdown-${operator}`,
+            "gd-button-primary",
+            "gd-button-small",
+            {
+                "button-dropdown": true,
+                "is-dropdown-open": opened,
+                "is-active": opened,
+            },
+        );
+
+        return (
+            <Button
+                title={title}
+                className={buttonClasses}
+                value={title}
+                onClick={this.handleOperatorDropdownButtonClick}
+                iconLeft={`icon-${getOperatorIcon(operator)}`}
+                iconRight={opened ? "icon-navigateup" : "icon-navigatedown"}
+            />
+        );
+    }
+
+    private handleOperatorDropdownClosed = () => {
+        this.closeOperatorDropdown();
+    };
+
+    private handleOperatorSelected = (operator: string) => {
+        this.closeOperatorDropdown();
+
+        if (operator === Operators.ALL) {
+            this.props.onSelect(undefined);
+        } else {
+            this.props.onSelect(operator);
+        }
+    };
+
+    private closeOperatorDropdown = () => {
+        this.setState({ opened: false });
+    };
+
+    private handleOperatorDropdownButtonClick = () => {
+        this.setState({ opened: !this.state.opened });
```

Comment:
Please use the `this.setState(state => ...)` syntax here (see my other comment on similar issue).